eng/smoke-tests/pom.xml

```diff
@@ -24,7 +24,7 @@
     <dependency>
       <groupId>com.microsoft.azure</groupId>
       <artifactId>azure-cosmos</artifactId>
-      <version>3.1.0</version>
+      <version>3.3.0</version>
```

Comment:
@kushagraThapar does it need to be 3.2.1?
Also in-general other pattern followed is to use SNAPSHOT suffix. This was followed earlier for cosmos. Is this the new strategy?

Reply:
@kirankumarkolli and @kushagraThapar  I'd looked at history of the version updates and followed the same update pattern. This is one of several PRs dealing with version increment and there's no real continuity as each one seemed to increment differently. That being said, for these binaries if we're going to change the increment pattern this would probably be a good time to do it.

Reply:
For general Java versioning guidance see https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#java, we generally don't use SNAPSHOT versions because of its viral nature as well as it adds reliability issues with always taking the latest snapshot version.

Reply:
@kirankumarkolli and/or @kushagraThapar are we good with the version number change?

Reply:
In short: Let's back this out. 

Smoke tests are a special case. They are supposed to simulate customer code (i.e. not tied as tightly into the rest of the build/engineering system) and exercise multiple SDK libraries within the same process. 

To get these smoke tests to where we can use them as a quality gate I think this change is appropriate so long as it matches the version of the package that is built at the current state of the repository (as it does in this change).

After speaking with @JimSuplizio I think this change moves us incrementally in the right direction, except it's going to turn the Smoke Test pipeline red until I add a few more changes to that pipeline. To keep us unblocked we can back this change out and do that work as part of another work item.